tft.py

```diff
@@ -0,0 +1,213 @@
+            champ_obj["stats"]["damage"] = record.getv("BaseDamage")
+            champ_obj["stats"]["armor"] = record.getv("baseArmor")
+            champ_obj["stats"]["magicResist"] = record.getv("baseSpellBlock")
+            champ_obj["stats"]["critMultiplier"] = record.getv("critDamageMultiplier")
+            champ_obj["stats"]["critChance"] = record.getv("baseCritChance")
+            champ_obj["stats"]["attackSpeed"] = record.getv("attackSpeed")
+            champ_obj["stats"]["range"] = record.getv("attackRange") // 180
+
+            champ_obj["ability"] = {}
+            champ_obj["ability"]["name"] = champ.getrv("87a69a5e")
+            champ_obj["ability"]["desc"] = champ.getrv("bc4f18b3")
+            champ_obj["ability"]["icon"] = champ.getv("mPortraitIconPath")
+            champ_obj["ability"]["variables"] = []
+
+            spellName = record.getv("spellNames")[0]
+            if "/" in spellName:
+                spellName = spellName[spellName.find("/") + 1 :]
+
+            ability = [x.getv("mSpell") for x in tft_bin.entries if x.type == "SpellObject" and x.getv("mScriptName") == spellName][0]
+            for value in ability.getv("mDataValues"):
+
+                ability_data = {}
+                ability_data["name"] = value.getv("mName")
+                ability_data["values"] = value.getv("mValues")
+
+                champ_obj["ability"]["variables"].append(ability_data)
+
+            champs[champ_id] = champ_obj
+
+        return champs
+
+    def parse_traits(self, map22):
+        trait_collection = [x for x in map22.entries if x.type == "TftTraitData"]
+        traits = []
+
+        for trait in trait_collection:
+            if "Template" in trait.getv("mName"):
+                continue
+
+            trait_obj = {}
+            trait_obj["internal"] = {}
+            trait_obj["internal"]["hash"] = trait.path.hex
+            trait_obj["internal"]["name"] = trait.getv("mName")
+            trait_obj["name"] = trait.getrv("c3143d66")
+            trait_obj["desc"] = trait.getrv("765f18da")
+            trait_obj["icon"] = trait.getv("mIconPath")
+
+            for trait_set in trait.getv("mTraitSets"):
+                trait_effects = {}
+                trait_effects["minUnits"] = trait_set.getv("mMinUnits")
+                trait_effects["maxUnits"] = trait_set.getv("mMaxUnits")
+                trait_effects["variables"] = []
+
+                if not trait_set.get("effectAmounts"):
+                    continue
+
+                for effect in trait_set.getv("effectAmounts"):
+                    trait_effect = {}
+                    trait_effect["name"] = effect.getv("name")
+                    trait_effect["value"] = effect.getv("value")
+
+                    trait_effects["variables"].append(trait_effect)
+
+            trait_obj["effects"] = trait_effects
+            traits.append(trait_obj)
+
+        return traits
+
+
+if __name__ == "__main__":
+    Transformer()
```

Comment:
The constructor should not do all the stuff.
You should use `argparse` to parse options (output path, etc.), then (if a class is needed) create an instance with those parameters (creating the instance not do the processing itself), then call a processing method.